### ISSUE TITLE
SotA 20 North Knalga: Lua error on start (GNA #25615)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1640]
**[Original submission](https://gna.org/bugs/index.php?25615)** by wedge009 on 2017-03-23

```

20170323 22:35:14 error scripting/lua: lua/backwards-compatibility.lua:11: bad argument #1 to 'tovconfig' (WML table expected, got number)

stack traceback:

	[C]: in field 'tovconfig'

	lua/backwards-compatibility.lua:11: in field 'fire'

	lua/helper.lua:244: in field 'add_ai_component'

	ai/micro_ais/micro_ai_helper.lua:96: in field 'add_CAs'

	ai/micro_ais/micro_ai_helper.lua:200: in field 'micro_ai_setup'

	ai/micro_ais/micro_ai_wml_tag.lua:51: in local 'cmd'

	lua/wml-utils.lua:145: in field 'handle_event_commands'

	lua/wml-flow.lua:6: in function <lua/wml-flow.lua:5>

```

Rated minor because there doesn't appear to be any serious side-effects (so far). The Lua file name implies it's only there for compatibility reasons, but I don't know what it's actually supposed to do.

Doesn't appear to be OS-specific, a clean load (via debug choose_level still has the error).

(Reproduced on All)
Release: 1.13.7+dev
Priority: 5 - Normal
Severity: 2 - Minor

**Comment by wesnoth-bugs** [comment]
**Modified on 2017-03-23**

**celticminstrel wrote:**

> I'd say the "backwards compatibility" appearing in the stack trace is misleading.
> 
> 
> 
> In any case, this should hopefully be fixed by f45d7951347f.

**celticminstrel changed status:** None -> Ready For Test

**Comment by wesnoth-bugs** [comment]
**Modified on 2017-03-23**

**wedge009 changed status:** Ready For Test -> Fixed

**wedge009 wrote:**

> Seems to be. Thanks.
